tests/unit/via/services/pdf_url_test.py

```diff
@@ -0,0 +1,137 @@
+                minute=30,
+                second=21,
+                tzinfo=timezone.utc,
+            ),
+        )
+
+
+class TestPDFURLBuilder:
+    @pytest.mark.parametrize(
+        "file_details,url",
+        (
+            (
+                {"file_id": "FILE_ID"},
+                "http://example.com/google_drive/FILE_ID/proxied.pdf"
+                "?url=http%3A%2F%2Fgdrive%2Fdocument.pdf",
+            ),
+            (
+                {"file_id": "FILE_ID", "resource_key": None},
+                "http://example.com/google_drive/FILE_ID/proxied.pdf"
+                "?url=http%3A%2F%2Fgdrive%2Fdocument.pdf",
+            ),
+            (
+                {"file_id": "FILE_ID", "resource_key": "RESOURCE_KEY"},
+                "http://example.com/google_drive/FILE_ID/RESOURCE_KEY/proxied.pdf"
+                "?url=http%3A%2F%2Fgdrive%2Fdocument.pdf",
+            ),
+        ),
+    )
+    def test_google_file_url(
+        self, svc, google_drive_api, secure_link_service, file_details, url
+    ):
+        google_drive_api.parse_file_url.return_value = file_details
+
+        pdf_url = svc.get_pdf_url("http://gdrive/document.pdf")
+
+        secure_link_service.sign_url.assert_called_once_with(url)
+        assert pdf_url == secure_link_service.sign_url.return_value
+
+    def test_nginx_file_url(self, google_drive_api, svc):
+        google_drive_api.parse_file_url.return_value = None
+
+        pdf_url = svc.get_pdf_url("http://nginx/document.pdf")
+
+        svc.nginx_signer.sign_url.assert_called_once_with(
+            "http://nginx/document.pdf", nginx_path="/proxy/static/"
+        )
+        assert pdf_url == svc.nginx_signer.sign_url.return_value
+
+    @pytest.fixture
+    def svc(
+        self,
+        google_drive_api,
+        secure_link_service,
+        pyramid_request,
+        NGINXSigner,
+    ):
+        return PDFURLBuilder(
+            google_drive_api,
+            secure_link_service,
+            pyramid_request.route_url,
+            NGINXSigner,
+        )
+
+
+class TestFactory:
+    def test_it(
+        self,
+        pyramid_request,
+        pyramid_settings,
+        PDFURLBuilder,
+        NGINXSigner,
+        google_drive_api,
+        secure_link_service,
+    ):
+        svc = factory(sentinel.context, pyramid_request)
+
+        NGINXSigner.assert_called_once_with(
+            nginx_server=pyramid_settings["nginx_server"],
+            secret=pyramid_settings["nginx_secure_link_secret"],
+        )
+        PDFURLBuilder.assert_called_once_with(
+            google_drive_api=google_drive_api,
+            secure_link_service=secure_link_service,
+            route_url=pyramid_request.route_url,
+            nginx_signer=NGINXSigner.return_value,
+        )
+        assert svc == PDFURLBuilder.return_value
+
+    @pytest.fixture
+    def PDFURLBuilder(self, patch):
+        return patch("via.services.pdf_url.PDFURLBuilder")
+
+
+@pytest.fixture
+def NGINXSigner(patch):
```

Comment:
```suggestion
def _NGINXSigner(patch):
```

I think our style is to exactly match the class name.

Reply:
`_NGINXSigner` clashes with the import of the real clash, not sure which one to alias or if having `_NGINXSigner` as a class fixture in two classes (instead of a global one) is better.

Reply:
Used a different name to avoid the name clash, but to something less subtle than removing the underscore `PatchedNGINXSigner`